### PR TITLE
Add violation_time_limit_seconds field to nrql conditions

### PR DESCRIPTION
**Pull request by jaymcgrath**
Prerequisite for https://github.com/newrelic/terraform-provider-newrelic/pull/1015
Update client to add support for `violation_time_limit_seconds` field in nerdgraph NRQL conditions. This field is now the preferred way to express violationTTL in nerdgraph, as it now supports a full range of seconds values from 300 (5 minutes) to 2592000 (30 days).  Adding support for `violation_time_limit_seconds` is necessary to [deprecate the enum-based `violation_time_limit` field in the terraform provider](https://github.com/newrelic/terraform-provider-newrelic/pull/1015).

### CODE REVIEW

**Comment by jaymcgrath** [comment]
It looks like there are some issues with the CI and github actions?  Most of the checks are failing on the `Add GOBIN to PATH` step: 
```
Error: Unable to process command '::add-path::/home/runner/go/bin' successfully.
Error: The `add-path` command is disabled. Please upgrade to using Environment Files or opt into unsecure command execution by setting the `ACTIONS_ALLOW_UNSECURE_COMMANDS` environment variable to `true`. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

```

**Comment by jaymcgrath** [comment]
Hi @ctrombley! Pending the CI fix for this repo, can we wait until after 11/30 to review/merge this? It's a prequisite for https://github.com/newrelic/terraform-provider-newrelic/pull/1015, which involves some change management.

Thanks!

**Review by ahornerr — APPROVED** [review]

**Pull request merged**
